### ISSUE TITLE
After a new PV is added to the VG, the logical volume of stripe type still fails to be created

### ISSUE THREAD

**Issue by tangtx3** [alibaba/open-local#5]
### I tried to create a 3Gi 'striped' PVC, but failed to create it, log as follows:
> E0811 01:33:55.208394       1 controller.go:920] error syncing claim "90e63057-6723-40fd-83d2-39199018c149": 
failed to provision volume with StorageClass "test": rpc error: code = Unknown desc = Create Lvm with error rpc error: code = Internal desc = failed to create lv: Failed to run cmd: **/nsenter --mount=/proc/1/ns/mnt --ipc=/proc/1/ns/ipc --net=/proc/1/ns/net --uts=/proc/1/ns/uts  lvcreate -n disk-90e63057-6723-40fd-83d2-39199018c149 -L 3221225472b -W y -y -i 2 vg_test with out:   Using default stripesize 64.00 KiB.**
  Insufficient suitable allocatable extents for logical volume disk-90e63057-6723-40fd-83d2-39199018c149: 494 more required, with error: exit status 5

### PV info
> [root@node-219 ~]# pvs -S vg_name=vg_test
  PV         VG            Fmt  Attr PSize   PFree
  /dev/sdb   vg_test lvm2 a--  <20.00g  540m
  /dev/sdc   vg_test lvm2 a--  <30.00g  <10.54g

### After I added a 30G PV(**/dev/sdd**) to VG, but still failed to create it, log as follows:
> E0811 01:33:55.208394       1 controller.go:920] error syncing claim "4b3a6936-8526-41f0-aa3d-f9e04bad4f3f": 
failed to provision volume with StorageClass "test": rpc error: code = Unknown desc = Create Lvm with error rpc error: code = Internal desc = failed to create lv: Failed to run cmd: **/nsenter --mount=/proc/1/ns/mnt --ipc=/proc/1/ns/ipc --net=/proc/1/ns/net --uts=/proc/1/ns/uts  lvcreate -n disk-4b3a6936-8526-41f0-aa3d-f9e04bad4f3f -L 3221225472b -W y -y -i 3 vg_test with out:   Using default stripesize 64.00 KiB.**
  Insufficient suitable allocatable extents for logical volume disk-4b3a6936-8526-41f0-aa3d-f9e04bad4f3f: 238 more required, with error: exit status 5

### PV info
> [root@node-219 ~]# pvs -S vg_name=vg_test
  PV         VG            Fmt  Attr PSize   PFree
  /dev/sdb   vg_test lvm2 a--  <20.00g  540m
  /dev/sdc   vg_test lvm2 a--  <30.00g  <10.54g
  /dev/sdd   vg_test lvm2 a--  <30.00g  <30.00g

### Then I looked up the problem and found it was a bug. No matter how many new PVS I added, I couldn't create a 'striped' volume. It depends on the PV with the lowest free capacity.
The code is as follows:
[https://github.com/alibaba/open-local/blob/e6feccc9591a755810aab375723aa4514dc6a1f1/pkg/csi/server/commands.go#L100](https://github.com/alibaba/open-local/blob/e6feccc9591a755810aab375723aa4514dc6a1f1/pkg/csi/server/commands.go#L100)

### If you're creating a 'striped' volume, I don't think it's right to get all the PVS in the VG as the '-i' parameter. I've optimized the code. I hope it helps.

```
	if striping {
		pvCount, err := getRequiredPVNumber(vg, size)
		if err != nil {
			return "", err
		}
		if pvCount == 0 {
			return "", fmt.Errorf("could not create `striping` logical volume, not enough space")
		}
		args = append(args, "-i", strconv.Itoa(pvCount))
	}
```
```
func getRequiredPVNumber(vgName string, lvSize uint64) (int, error) {
	pvs, err := ListPV(vgName)
	if err != nil {
		return 0, err
	}
	// calculate pv count
	pvCount := len(pvs)
	for pvCount > 0 {
		avgPvRequest := lvSize / uint64(pvCount)
		for num, pv := range pvs {
			if pv.FreeSize < avgPvRequest {
				pvs = append(pvs[:num], pvs[num+1:]...)
			}
		}

		if pvCount == len(pvs) {
			break
		}
		pvCount = len(pvs)
	}
	return pvCount, nil
}

```

**Comment by TheBeatles1994** [comment]
Thanks for your issue and detailed improvement! You can create a pull request to fix this!

**Comment by TheBeatles1994** [comment]
Fixed by #6
